modules/transport-netty4/src/test/java/org/elasticsearch/http/netty4/Netty4HttpServerTransportTests.java

```diff
@@ -343,14 +356,33 @@ public void dispatchBadRequest(final RestChannel channel, final ThreadContext th
                 dispatcher,
                 clusterSettings,
                 new SharedGroupFactory(Settings.EMPTY)
-            )
+            ) {
+                @Override
+                public ChannelHandler configureServerChannelHandler() {
+                    return new HttpChannelHandler(this, handlingSettings) {
+                        @Override
+                        protected void initChannel(Channel ch) throws Exception {
+                            super.initChannel(ch);
+                            ch.pipeline().addBefore("pipelining", "assert-throttling", new ChannelOutboundHandlerAdapter() {
+                                @Override
+                                public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise) throws Exception {
+                                    assertTrue("handler should throttle to only write into writable channels", ctx.channel().isWritable());
```

Comment:
This verifies one side of the equation, that we do not write into non-writeable channels. Does the test also ensure that the channel indeed goes non-writeable and can we verify that somewhere, perhaps by adding another handling after the pipelining handler that snoops this?

Reply:
Looks good, thanks.